### PR TITLE
Improvement/bb 49/change probe server variable name

### PR DESCRIPTION
**Pull request by naren-rajendran**
- improvement: BB-53 use updated probe server in GC process.
- improvement: BB-49 change probe server variable name and use the updated named in replication processes.

### CODE REVIEW

File: extensions/gc/service.js [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
+    DEFAULT_LIVE_ROUTE,
+    DEFAULT_READY_ROUTE,
+} = require('arsenal').network.probe.ProbeServer;
 
 const GarbageCollector = require('./GarbageCollector');
-
+const { sendSuccess, sendError, startProbeServer } = require('../../lib/util/probe');
+const { initManagement } = require('../../lib/management');
 const config = require('../../lib/Config');
+const { process } = require('ipaddr.js');
```
Comment (miniscruff) [inline extensions/gc/service.js]: I don't see this being used

File: extensions/gc/service.js [hunk 2]
```diff
@@ -39,22 +55,22 @@ function initAndStart() {
...
-            if (garbageCollector.isReady()) {
-                return true;
-            }
-            log.error('GarbageCollector is not ready!');
-            return false;
-        });
-        logger.info('Starting HealthProbe server');
-        healthServer.start();
-        garbageCollector.start(err => {
+        startProbeServer(config.healthcheckServer, (err, probeServer) => {
```
Comment (miniscruff) [inline extensions/gc/service.js]: Not sure how much more work it is but using a new config object to match the others might make sense here

**Review by miniscruff — CHANGES_REQUESTED** [review]

File: extensions/gc/service.js (hunk 2 above)
Comment (naren-rajendran) [reply extensions/gc/service.js]: new config object?

File: extensions/gc/service.js (hunk 1 above)
Comment (naren-rajendran) [reply extensions/gc/service.js]: how did that squeeze in?

File: extensions/gc/service.js (hunk 2 above)
Comment (miniscruff) [reply extensions/gc/service.js]: yea like config.extensions.gc.probeServer to match the replication ones

File: docs/replication-metrics.md [hunk 3]
```diff
@@ -11,11 +11,11 @@ Under `conf/config.json` you can specify the probe server settings.
 
 > Currently SSL/TLS is not supported
 
-When starting processes you can disable the probe server by setting an environment
-variable.
+When starting processes you can enable/disable metrics handling by setting an
+environment variable.
```
Comment (miniscruff) [inline docs/replication-metrics.md]: Can you mention it defaults to OFF

**Review by nicolas2bert — APPROVED** [review]

**Review by miniscruff — APPROVED** [review]

**Pull request merged**